### ISSUE TITLE
Auto-Setting upstream container version in build.py for no container build

### ISSUE THREAD

**Issue by jishminor** [triton-inference-server/server#4045]
**Description**
When running a no container build using build.py, upstream image is not set resulting in failed docker image pulls for the desired backends.

**Triton Information**
r22.02

Are you using the Triton container or did you build it yourself?
Building myself using no container build

**To Reproduce**
Steps to reproduce the behavior.
Running on an x86 ubuntu 20.04 machine. Command being run to build is:
```bash
./build.py --cmake-dir=$HOME/code/server/build --build-dir=/tmp/citritonbuild --no-container-build --enable-logging --enable-stats --enable-tracing --enable-metrics --endpoint=http --endpoint=grpc --backend=pytorch --backend=tensorflow2 --backend=tensorflow1 --extra-backend-cmake-arg=tensorflow2:TRITON_TENSORFLOW_INSTALL_EXTRA_DEPS=ON
```

**Expected behavior**
In `build.py` `FLAGS.upstream_container_version` only gets set based on the version map if building with the container build. Line reference [here](https://github.com/triton-inference-server/server/blob/f6ad9caf5ebe5c7a8b93f176df9927af42d588c5/build.py#L1737). Is there a reason this is not also done for the no container build?

If I build with passing the flag `--upstream-container-version=22.02`, the build completes fine.

**Comment by dzier** [comment]
This appears to be an oversight. Filing an internal bug.

**Comment by deadeyegoodwin** [comment]
Fixed with #4045
